File: Sources/Classes/RSConfig.m
//
//  RSConfig.m
//  RSSDKCore
//
//  Created by Arnab Pal on 17/10/19.
//  Copyright © 2019 RSlabs. All rights reserved.
//

#import "RSConfig.h"
#import "RSConstants.h"
#import "RSLogger.h"

@implementation RSConfig

- (instancetype)init {
    self = [super init];
    if (self) {
        _dataPlaneUrl = nil;
        _dataResidencyServer = US;
        _flushQueueSize = RSFlushQueueSize;
        _dbCountThreshold = RSDBCountThreshold;
        _sleepTimeout = RSSleepTimeout;
        _logLevel = RSLogLevelError;
        _configRefreshInterval = RSConfigRefreshInterval;
        _sessionInActivityTimeOut = RSSessionInActivityDefaultTimeOut;
        _trackLifecycleEvents = RSTrackLifeCycleEvents;
        _recordScreenViews = RSRecordScreenViews;
        _enableBackgroundMode = RSEnableBackgroundMode;
        _automaticSessionTracking = RSAutomaticSessionTracking;
        _collectDeviceId = RSCollectDeviceId;
        _controlPlaneUrl = RSControlPlaneUrl;
        _factories = [[NSMutableArray alloc] init];
        _customFactories = [[NSMutableArray alloc] init];
        _gzip = RSGzipStatus;
        _dbEncryption = nil;
    }
    return self;
}

@end

File: Rudder/Rudder/Domain/Models/RSConfig.swift
//
//  RSConfig.swift
//  Rudder
//
//  Created by Pallab Maiti on 04/08/21.
//  Copyright © 2021 Rudder Labs India Pvt Ltd. All rights reserved.
//

import Foundation

@objc open class RSConfig: NSObject {
    let dataPlaneUrl: String
    let flushQueueSize: Int
    let dbCountThreshold: Int
    let sleepTimeout: Int
    let logLevel: Int
    let configRefreshInterval: Int
    let trackLifecycleEvents: Bool
    let recordScreenViews: Bool
    let controlPlaneUrl: String
    let factories: [RSIntegrationFactory]
    
    public override init() {
        dataPlaneUrl = Constants.RSDataPlaneUrl
        flushQueueSize = Constants.RSFlushQueueSize
        dbCountThreshold = Constants.RSDBCountThreshold
        sleepTimeout = Constants.RSSleepTimeout
        logLevel = 1 //TODO: Need to replace with RSLogger
        configRefreshInterval = Constants.RSConfigRefreshInterval
        trackLifecycleEvents = Constants.RSTrackLifeCycleEvents
        recordScreenViews = Constants.RSRecordScreenViews
        controlPlaneUrl = Constants.RSControlPlaneUrl
        factories = [RSIntegrationFactory]()
    }
    
    init(dataPlaneUrl: String, flushQueueSize: Int, dbCountThreshold: Int, sleepTimeout: Int, logLevel: Int, configRefreshInterval: Int, trackLifecycleEvents: Bool, recordScreenViews: Bool, controlPlaneUrl: String) {
        self.dataPlaneUrl = dataPlaneUrl
        self.flushQueueSize = flushQueueSize
        self.dbCountThreshold = dbCountThreshold
        self.sleepTimeout = sleepTimeout
        self.logLevel = logLevel
        self.configRefreshInterval = configRefreshInterval
        self.trackLifecycleEvents = trackLifecycleEvents
        self.recordScreenViews = recordScreenViews
        self.controlPlaneUrl = controlPlaneUrl
        self.factories = [RSIntegrationFactory]()
    }
}
